library/block/block_content_registry/src/main/java/org/quiltmc/qsl/block/content/registry/api/EnchantingBoosters.java

```diff
@@ -0,0 +1,160 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.block.content.registry.api;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.mojang.datafixers.util.Either;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+/**
+ * The class responsible for managing {@link EnchantingBooster}s
+ */
+public class EnchantingBoosters {
+	private static final BiMap<Identifier, EnchantingBoosterType> TYPES = HashBiMap.create();
+
+	/**
+	 * The codec for {@link EnchantingBoosterType}s
+	 */
+	public static Codec<EnchantingBoosterType> TYPE_CODEC = Identifier.CODEC.flatXmap(id -> {
+		EnchantingBoosterType type = TYPES.get(id);
+		return type != null ? DataResult.success(type) : DataResult.error("Unknown enchanting booster type: " + id);
+	}, type -> {
+		Identifier identifier = TYPES.inverse().get(type);
+		return identifier != null ? DataResult.success(identifier) : DataResult.error("Unknown enchanting booster type");
+	});
+
+	private static final Codec<Either<Either<Float, Identifier>, EnchantingBooster>> EITHER_CODEC = Codec.either(
+			Codec.either(Codec.FLOAT, Identifier.CODEC),
+			TYPE_CODEC.dispatch(EnchantingBooster::getType, EnchantingBoosterType::codec)
+	);
+	/**
+	 * The codec for an {@link EnchantingBooster}
+	 */
+	public static final Codec<EnchantingBooster> CODEC = EITHER_CODEC
+			.flatXmap(either ->
+						either.<DataResult<EnchantingBooster>>map(floatId ->
+							floatId.map(f -> DataResult.success(new ConstantBooster(f)), id -> {
+								EnchantingBoosterType type = TYPES.get(id);
+								if (type == null) {
+									return DataResult.error("Unknown Booster Type: " + id);
+								}
+
+								return type.isSimple() ? DataResult.success(type.simple()) : DataResult.<EnchantingBooster>error("Booster Type: " + id + " is not simple. Please fully specify it.");
+							}),
+							DataResult::success),
+					enchantingBooster -> {
+						if (enchantingBooster instanceof ConstantBooster constant) {
+							return DataResult.success(Either.left(Either.left(constant.value())));
+						}
+
+						if (enchantingBooster.getType().isSimple() && enchantingBooster.getType().simple() == enchantingBooster) {
+							return DataResult.success(Either.left(Either.right(TYPES.inverse().get(enchantingBooster.getType()))));
+						}
+
+						return DataResult.success(Either.right(enchantingBooster));
+					});
+
+
+	/**
+	 * The interface in charge of calculating the enchanting boost value
+	 */
+	public interface EnchantingBooster {
```

Comment:
Maybe move this interface to its own file?

Reply:
yeah i should probably do that